testing/deployer/topology/default_versions.go

```diff
@@ -5,9 +5,10 @@
 
 package topology
 
+//TODO @sarah.alsmiller figure out if we can delete this
 const (
```

Comment:
`testing/deployer` was originally going to be a standard module rather than a submodule, so you could reuse it elsewhere to write tests (like in consul-dataplane). Because of that it has these default versions embedded in it, so each test didn't have to always pick specific versions for everything.

The `sprawltest` test is a unit test that tests the deployer stuff works by itself (as a library) so it needs versions to use for that (but i don't think we actually actively run those tests in CI)

Since this ended up as a submodule with only one consumer we could switch this to have the topology config validation fail if any of the image names are empty, thus requiring the user to move 100% of the version selection up front to each test itself rather than having any defaults at all down here.

Reply:
Since I cloned a huge chunk of this library from my "spin up local test environments" tool, I was holding out hope that the library could be used outside of tests to run local dev environments for people, but I suspect practically speaking most of the devs just use `consul-k8s` for weird setups and don't really have strong needs for "bare metal" setups.

Reply:
@rboyer thanks a bunch for the context - and yeah totally agree on both points; I’d love to see every required version defaulted in one place or not at all (provided by test env explicitly); same with Docker build args for versions.

And yeah, this seems valuable for bare metal testing but I do think most folks lean heavily on k8s for most env-agnostic tests.